src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3612,27 +3547,20 @@ public override SyntaxNode ObjectCreationExpression(SyntaxNode type, IEnumerable
             return SyntaxFactory.ObjectCreationExpression((TypeSyntax)type, CreateArgumentList(arguments), null);
         }
 
-        private ArgumentListSyntax CreateArgumentList(IEnumerable<SyntaxNode> arguments)
+        private static ArgumentListSyntax CreateArgumentList(IEnumerable<SyntaxNode> arguments)
         {
             return SyntaxFactory.ArgumentList(CreateArguments(arguments));
         }
 
-        private SeparatedSyntaxList<ArgumentSyntax> CreateArguments(IEnumerable<SyntaxNode> arguments)
+        private static SeparatedSyntaxList<ArgumentSyntax> CreateArguments(IEnumerable<SyntaxNode> arguments)
         {
-            return SyntaxFactory.SeparatedList(arguments.Select(AsArgument).Cast<ArgumentSyntax>());
+            return SyntaxFactory.SeparatedList(arguments.Select(AsArgument));
```

Comment:
Here's our general policy:  Code should be readable and maintainable.

That's what we should do up to the point that it's an actual problem for some other reason.  For example: perf.  But, we don't take the position that just because there might be a perf issue in one hot path that you should avoid this stuff everywhere.  Instead, we like to drive changes like that through data that demonstrates the problem and allows us to explain in teh code why we're doing something different.